### ISSUE TITLE
[Bug][zos_mount] Wrong type for parameter tag_ccsid of zos_mount module

### ISSUE THREAD

**Issue by dvdrbn-m9** [ansible-collections/ibm_zos_core#352]
##### SUMMARY

When using module zos_mount with options `tag_untagged: TEXT` & `tag_ccsid: 819` I receive this message:
```
  msg: Parameter verification failed
  stderr: Invalid argument "819" for type "str".
  stderr_lines: <omitted>
```

##### COMPONENT NAME

zos_mount

##### ANSIBLE VERSION

```
ansible [core 2.12.1]
  config file = /home/davidr/project/ansible/ansible.cfg
  configured module search path = ['/home/davidr/.ansible/plugins/modules', '/usr/share/ansible/plugins/modules']
  ansible python module location = /home/davidr/.local/lib/python3.10/site-packages/ansible
  ansible collection location = /home/davidr/.ansible/collections:/usr/share/ansible/collections
  executable location = /home/davidr/.local/bin/ansible
  python version = 3.10.5 (main, Jun  6 2022, 18:49:26) [GCC 12.1.0]
  jinja version = 3.1.2
  libyaml = True
```

##### SPECIFY ANSIBLE COLLECTION VERSION

1.4.0-beta.1

##### SPECIFY THE Z OPEN AUTOMATION UTILITIES VERSION

2021/08/05 22:12:58 CUT V1.2.0

##### ENVIRONMENT

z/OS v2.5

##### STEPS TO REPRODUCE

The following task should cause the issue.
```yaml
- name: Mount ZFS
  zos_mount:
    src: "ANSBL.TEST.ZFS"
    path: "/tmp/test"
    fs_type: ZFS
    mount_opts: RW
    state: mounted
    tag_untagged: TEXT
    tag_ccsid: 819
  register: mount_result
```

##### EXPECTED RESULTS

ZFS is mounted successfully with `TAG(TEXT,819)`. This does work with regular TSO MOUNT comand.

##### ACTUAL RESULTS

```
The full traceback is:
  File "/tmp/ansible_zos_mount_payload_5sxvugrk/ansible_zos_mount_payload.zip/ansible_collections/ibm/ibm_zos_core/plugins/modules/zos_mount.py", line 674, in run_module
  File "/tmp/ansible_zos_mount_payload_5sxvugrk/ansible_zos_mount_payload.zip/ansible_collections/ibm/ibm_zos_core/plugins/module_utils/better_arg_parser.py", line 787, in parse_args
    updated_value = handler.handle_arg()
  File "/tmp/ansible_zos_mount_payload_5sxvugrk/ansible_zos_mount_payload.zip/ansible_collections/ibm/ibm_zos_core/plugins/module_utils/better_arg_parser.py", line 163, in handle_arg
    self.contents = self._resolve_arg_type()
  File "/tmp/ansible_zos_mount_payload_5sxvugrk/ansible_zos_mount_payload.zip/ansible_collections/ibm/ibm_zos_core/plugins/module_utils/better_arg_parser.py", line 623, in _resolve_arg_type
    return self.type_handlers.get(self.arg_def.arg_type)(
  File "/tmp/ansible_zos_mount_payload_5sxvugrk/ansible_zos_mount_payload.zip/ansible_collections/ibm/ibm_zos_core/plugins/module_utils/better_arg_parser.py", line 214, in _dict_type
    updated_value = handler.handle_arg()
  File "/tmp/ansible_zos_mount_payload_5sxvugrk/ansible_zos_mount_payload.zip/ansible_collections/ibm/ibm_zos_core/plugins/module_utils/better_arg_parser.py", line 163, in handle_arg
    self.contents = self._resolve_arg_type()
  File "/tmp/ansible_zos_mount_payload_5sxvugrk/ansible_zos_mount_payload.zip/ansible_collections/ibm/ibm_zos_core/plugins/module_utils/better_arg_parser.py", line 623, in _resolve_arg_type
    return self.type_handlers.get(self.arg_def.arg_type)(
  File "/tmp/ansible_zos_mount_payload_5sxvugrk/ansible_zos_mount_payload.zip/ansible_collections/ibm/ibm_zos_core/plugins/module_utils/better_arg_parser.py", line 236, in _str_type
    raise ValueError('Invalid argument "{0}" for type "str".'.format(contents))
fatal: [lpar25]: FAILED! => changed=false 
  invocation:
    module_args:
      allow_uid: true
      attributes: null
      automove: AUTOMOVE
      automove_list: null
      fs_type: ZFS
      group: null
      mode: null
      mount_opts: RW
      owner: null
      path: /tmp/test
      persistent: null
      selevel: null
      serole: null
      setype: null
      seuser: null
      src: ANSBL.TEST.ZFS
      src_params: null
      state: mounted
      sysname: null
      tag_ccsid: 819
      tag_untagged: TEXT
      unmount_opts: NORMAL
      unsafe_writes: false
  msg: Parameter verification failed
  stderr: Invalid argument "819" for type "str".
  stderr_lines: <omitted>
```

**Comment by ddimatos** [comment]
@dvdrbn-m9 - thank you for reporting this, we will look into and update soon.

**Comment by ddimatos** [comment]
Just dropping a comment for later, [but this looks suspicious](https://github.com/ansible-collections/ibm_zos_core/blob/f84c04834f4697bae70ece0c07113465d2847f59/plugins/modules/zos_mount.py#L1161) @richp405

**Comment by richp405** [comment]
In the main body, ccsid is processed as an int, which is how it's defined.  However, in the 'main' routine, it was defined as a string, not an int.  So an un-decorated number (813 instead of "813") triggered the issue.

I've corrected it in 1402 patch, but am waiting for the 2 jobs in front of me to finish.

**Comment by ddimatos** [comment]
Forward ported to dev branch for future releases: https://github.com/ansible-collections/ibm_zos_core/pull/511

**Comment by ddimatos** [comment]
Fixed in 1.4.0-beta.2 https://github.com/ansible-collections/ibm_zos_core/pull/502
